include/kllvm/binary/deserializer.h

```diff
@@ -43,13 +43,36 @@ T read(It &ptr, It end) {
   return val;
 }
 
+template <typename It>
+uint64_t read_length(It &ptr, It end) {
+  uint64_t ret = 0;
+  auto should_continue = true;
+  auto steps = 0;
+
+  while (should_continue) {
```

Comment:
we should probably explicitly error out after the 9th byte, otherwise we might potentially read the entire file as a length field on malformed input, which could lead to the deserialization process hanging.